src/UserBundle/EventSubscriber/UserLoginEventSubscriber.php

```
@@ -50,6 +55,20 @@ public function onAuthenticationSuccess(AuthenticationSuccessEvent $event): void
         if (! $user->isVerified()) {
             throw new UserNotVerifiedException();
         }
+
+        $request = $this->requestStack->getMainRequest();
+        $resolved = $request?->attributes->get(HostRoutingListener::REQUEST_ATTR);
+
+        if ($resolved instanceof ResolvedHost && $resolved->isCustomDomain() && $resolved->company !== null) {
+            $companyId = $resolved->company->getId();
+            $belongs = $user->getCompanies()->exists(
+                static fn ($key, $company) => $company->getId()->equals($companyId)
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$key'. (undefined)

(UnusedFormalParameter)